src/common/core/coretype.js

```diff
@@ -303,10 +304,14 @@ define([
         //</editor-fold>
 
         //<editor-fold=Modified Methods>
+        this.isValidNodeThrow = function (node) {
```

Comment:
wouldn't it be better if the isValidNodeThrow functions would be 'helper' functions and not part of the class?